src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network Analysis/FindServiceArea/FindServiceArea.xaml.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright 2018 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.Tasks.NetworkAnalysis;
+using Esri.ArcGISRuntime.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace ArcGISRuntime.WPF.Samples.FindServiceArea
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Find service area (interactive)",
+        "Network Analysis",
+        "Demonstrates how to find services areas around a point using the ServiceAreaTask. A service area shows locations that can be reached from a facility based off a certain impedance [such as travel time]. Barriers can also be added which can effect the impedance by not letting traffic through or adding the time is takes to pass that barrier.",
+        "")]
+    public partial class FindServiceArea
+    {
+        // An object that defines the task to find service area around a facility.
+        private ServiceAreaTask _serviceAreaTask;
+
+        // An object that defines parameters for solving a service area task.
+        private ServiceAreaParameters _serviceAreaParameters;
+
+        // An overlay for displaying service areas to the map view.
+        private GraphicsOverlay _graphicsOverlay;
+
+        // A symbol for facilities.
+        private PictureMarkerSymbol _facilitySymbol;
+
+        // A symbol for service area outlines.
+        private SimpleLineSymbol _outline;
+
+        public FindServiceArea()
+        {
+            InitializeComponent();
+
+            // Create the map, graphics overlay, and sketch editor.
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Center the map on San Diego.
+            Map streetsMap = new Map(Basemap.CreateLightGrayCanvasVector());
+            streetsMap.InitialViewpoint = new Viewpoint(32.73, -117.14, 30000);
+            MyMapView.Map = streetsMap;
+
+            // Create graphics overlays for all of the elements of the map.
+            _graphicsOverlay = new GraphicsOverlay();
+            MyMapView.GraphicsOverlays.Add(_graphicsOverlay);
+
```

Comment:
Handle double-taps on the map view to prevent the map from zooming in when the user double-taps to end a sketch. If the user isn't sketching, the double-tap will zoom in as expected.
```
            // Handle double-tap events if the user is drawing a sketch.
            MyMapView.GeoViewDoubleTapped += (s, e) => 
            {
                // If the sketch editor complete command is enabled, a sketch is in progress.
                if (MyMapView.SketchEditor.CompleteCommand.CanExecute(null))
                {
                    // Set the event as handled.
                    e.Handled = true;
                }
            };
```